tests/Repository/IncentiveRepositoryTest.php

```diff
@@ -0,0 +1,217 @@
+{
+    private $em;
+    private $repo;
+
+    public function setup(): void
+    {
+        self::bootKernel();
+        $this->em = static::$container->get(EntityManagerInterface::class);
+        $this->repo = static::$container->get(IncentiveRepository::class);
+
+    }
+
+    /**
+     * @dataProvider paginationDataProvider
+     */
+    public function testOnsitePatientStatusPagination($start, $length, $resultCount, $resultParticipantId): void
+    {
+        $this->createIncentives();
+        $params = [];
+        $params['start'] = $start;
+        $params['length'] = $length;
+        $incentives = $this->repo->getOnSiteIncentives('PS_SITE_TEST', $params);
+        $this->assertEquals($resultCount, count($incentives));
+        $this->assertEquals($resultParticipantId, $incentives[0]['participantId']);
+
+    }
+
+    public function paginationDataProvider()
+    {
+        return [
+            [0, 2, 2, 'P000000000'],
+            [1, 1, 1, 'P000000001'],
+            [3, 1, 1, 'P000000003']
+        ];
+    }
+
+    /**
+     * @dataProvider dateFilterDataProvider
+     */
+    public function testOnsiteIncentivesDateFilters($startDate, $endDate, $resultCount): void
+    {
+        $this->createIncentives();
+        $params = [];
+        $params['startDate'] = $startDate;
+        $params['endDate'] = $endDate;
+        $incentives = $this->repo->getOnSiteIncentives('PS_SITE_TEST', $params);
+        $this->assertEquals($resultCount, count($incentives));
+
+        $params = [];
+        $params['startDateOfService'] = $startDate;
+        $params['endDateOfService'] = $endDate;
+        $incentives = $this->repo->getOnSiteIncentives('PS_SITE_TEST', $params);
+        $this->assertEquals($resultCount, count($incentives));
+    }
+
+    public function dateFilterDataProvider()
+    {
+        return [
+            ['2022-01-15', '2022-02-15', 2],
+            ['2022-01-15', '2022-03-15', 3],
+            ['2022-01-15', '2022-05-15', 5],
+            ['2022-06-15', '2022-07-15', 0],
+            ['2022-02-15', '', 4],
+            ['', '2022-05-15', 5]
+        ];
+    }
+
+    /**
+     * @dataProvider participantIdDataProvider
+     */
+    public function testOnsiteIncentivesParticipantIdLookup($participantId): void
+    {
+        $this->createIncentives();
+        $params = [];
+        $params['participantId'] = $participantId;
+        $incentives = $this->repo->getOnSiteIncentives('PS_SITE_TEST', $params);
+        $this->assertEquals($participantId, $incentives[0]['participantId']);
+    }
+
+    public function participantIdDataProvider()
+    {
+        return [
+            ['P000000000'],
+            ['P000000001'],
+            ['P000000002'],
+            ['P000000003'],
+            ['P000000004']
+        ];
+    }
+
+    public function getOnsiteIncentivesCount(): void
+    {
+        $this->createIncentives();
+        $count = $this->repo->getOnsiteIncentivesCount('PS_SITE_TEST', []);
```

Comment:
The method has multiple `if` statements (for participant ID and dates) but the test does not cover any of those options.  Please add test scenarios to cover the optional parameters.